File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //use linkedhashmap
        int[] res = {0,0};
        int indice = 0;
        Map<Integer,Integer> map = new LinkedHashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            
            // Check if the complement (target - nums[i]) exists in the map
            if (map.containsKey(diff)) {
                // If it exists, return the indices of the two numbers
                res[0] = map.get(diff);
                res[1] = i;
                break;
            }
            else{
                map.put(nums[i], i);
            }
        }
        return res;
    }
        
}

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        //main idea : number - complement present in dict or not.
        Dictionary<int ,int> hashMap = new Dictionary<int,int>();
        for(int i =0;i < nums.Length;i++)
        {
            int secondNumber = target - nums[i];
            if(hashMap.ContainsKey(secondNumber))
            {
                return new int[] {hashMap[secondNumber],i};
            }
            else
            {
                hashMap[nums[i]] = i;
            }
        }
return new int[] {};

        
    }
}